src/main/java/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,163 @@
+    private static final int ID_COLUMN = 1;
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM orders WHERE user_id=?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            return getOrdersFromResultSet(resultSet);
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Cant create Shopping Cart.", ex);
+        }
+    }
+
+    @Override
+    public Order update(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "UPDATE internet_shop.orders "
+                    + "SET user_id=? "
+                    + "WHERE order_id=?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, order.getUserId());
+            statement.setLong(2, order.getOrderId());
+            for (Product product : order.getProducts()) {
+                query = "INSERT INTO orders_products(order_id, product_id) "
+                        + "VALUES (?,?)";
+                statement = connection.prepareStatement(query);
+                statement.setLong(1, order.getOrderId());
+                statement.setLong(2, product.getProductId());
+                statement.executeUpdate();
+            }
+            return order;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Cant update Shopping Cart.", ex);
+        }
+    }
+
+    @Override
+    public Order create(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO orders(user_id) VALUE (?)";
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                Long orderId = resultSet.getLong(ID_COLUMN);
+                order.setOrderId(orderId);
+                return update(order);
+            }
+
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Cant create Shopping Cart.", ex);
+        }
+        return null;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM orders WHERE order_id=?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            return getOrdersFromResultSet(resultSet).stream().findFirst();
```

Comment:
it would be better to make method which returns just one order from result set, and you can use it inside loop in method getAll()